diff --git MODULE.bazel MODULE.bazel
index 8b858e553..3a2caca01 100644
--- MODULE.bazel
+++ MODULE.bazel
@@ -46,7 +46,7 @@ bazel_dep(name = "rules_license", version = "1.0.0")
 bazel_dep(name = "rules_pkg", version = "1.0.1")
 bazel_dep(name = "rules_python", version = "1.0.0")
 
-bazel_dep(name = "rules_rust", version = "0.56.0", dev_dependency = True)
+bazel_dep(name = "rules_rust", version = "0.56.0")
 bazel_dep(name = "rules_ruby", version = "0.17.3", dev_dependency = True)
 
 # Workaround for https://github.com/bazelbuild/bazel-central-registry/issues/4230
@@ -148,7 +148,7 @@ pip = use_extension("@rules_python//python/extensions:pip.bzl", "pip", dev_depen
 
 use_repo(pip, "protobuf_pip_deps")
 
-crate = use_extension("@rules_rust//crate_universe:extension.bzl", "crate", dev_dependency = True)
+crate = use_extension("@rules_rust//crate_universe:extension.bzl", "crate")
 crate.spec(
     package = "googletest",
     version = ">0.0.0",
diff --git rust/bazel/aspects.bzl rust/bazel/aspects.bzl
index ec7a17283..f8a04cb8a 100644
--- rust/bazel/aspects.bzl
+++ rust/bazel/aspects.bzl
@@ -12,8 +12,6 @@ load("//bazel/common:proto_common.bzl", "proto_common")
 load("//bazel/common:proto_info.bzl", "ProtoInfo")
 load("//bazel/private:cc_proto_aspect.bzl", "cc_proto_aspect")
 
-visibility(["//rust/...", "//third_party/crubit/rs_bindings_from_cc/..."])
-
 CrateMappingInfo = provider(
     doc = "Struct mapping crate name to the .proto import paths",
     fields = {